packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -342,8 +342,9 @@ module.exports = (log, db, Password, verifierVersion, customs, mailer) => {
           throw errors.unverifiedSession();
         }
 
-        const result = await db.recoveryKeyExists(uid);
-        if (!result.exists) {
+        const keyForUid = await db.recoveryKeyExists(uid);
+
+        if (!keyForUid.exists) {
```

Comment:
Q: is it possible that `db.recoveryKeyExists()` could return `null`/`undefined`?
(and if so, would that cause unexpected errors when trying to access `.exists` on an `undefined` object, etc.)

Reply:
Return type _should_ be boolean, but I'm all for guarding against a `null` or `undefined` return type.

Reply:
The implementation here matches previous use by @vbudhram. I think that if `keyForUid` is `null` or `undefined`, we'll just get a falsy value for `!keyForUid.exists`, so this should be alright and not throw errors?